src/modules/hip/kernel/pre_emphasis_filter.hpp

```diff
@@ -0,0 +1,74 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+__device__ void pre_emphasis_filter_hip_compute(d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8, float4 *coeff_f4)
+{
+    dst_f8->f4[0] = src1_f8->f4[0]  - *coeff_f4 * src2_f8->f4[0];
+    dst_f8->f4[1] = src1_f8->f4[1]  - *coeff_f4 * src2_f8->f4[1];
+}
+
+__global__ void pre_emphasis_filter_tensor(float *srcPtr,
+                                           uint srcStride,
+                                           float *dstPtr,
+                                           uint dstStride,
+                                           Rpp32s *srcLengthTensor,
+                                           float *coeffTensor,
+                                           RpptAudioBorderType borderType)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x >= srcLengthTensor[id_z])
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStride) + id_x;
+    uint dstIdx = (id_z * dstStride) + id_x;
+
+    if(id_x == 0)
+    {
+        if(borderType == RpptAudioBorderType::ZERO)
+            dstPtr[dstIdx] = srcPtr[srcIdx];
+        else
+        {
+            float border = (borderType == RpptAudioBorderType::CLAMP) ? srcPtr[srcIdx] : srcPtr[srcIdx + 1];
+            dstPtr[dstIdx] = srcPtr[srcIdx] - coeffTensor[id_z] * border;
+        }
+    }
+
+    float4 coeff_f4 = (float4)coeffTensor[id_z];
```

Comment:
In all recent HIP kernels, we are using static_cast instead of c style casting
So please change this to
```
    float4 coeff_f4 = static_cast<float4>(coeffTensor[id_z]);
```

Reply:
Done